packages/react-native/tests/storage/chunkifier.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { WritableStream } from 'web-streams-polyfill';
+import { Chunk, ChunkifierSink, ChunkSplitter } from '../../src/storage/Chunkifier';
+import { blackholeChunkSink } from '../_helpers/blackholeChunkSink';
+import { splitToEnd } from '../_helpers/chunks';
+import { dataStream, randomString, readToEnd } from '../_helpers/generators';
+import { memoryChunkSink } from '../_helpers/memoryChunkSink';
+
+function charSplitter(char: string): ChunkSplitter<string> {
+    return (chunk) => {
+        const index = chunk.indexOf(char);
+        if (index === -1) {
+            return [chunk];
+        }
+        return [chunk.substring(0, index), chunk.substring(index + 1)];
+    };
+}
+
+function noopSplitter<W extends Chunk>(): ChunkSplitter<W> {
+    return (c: W) => [c];
+}
+
+describe('ChunkifierSink', () => {
+    it('should call splitter function with every chunk', async () => {
```

Comment:
is it worth adding a case where splitter should not be called?

Reply:
Added.